File: Java/172_remove_element.java
public class Solution {
    /*
     * @param A: A list of integers
     * @param elem: An integer
     * @return: The new length after remove
     */
    public int removeElement(int[] A, int elem) {
        int fast = 0, slow = 0;
        int n = A.length;
        int ret = 0;
        while(fast < n){
            if(A[fast] != elem) {
                ++ret;
                A[slow++] = A[fast];
            }
            ++fast;
        }
        return ret;
    }
}

File: cpp/172_remove_element.cpp
// Time:  O(n)
// Space: O(1)

class Solution {
  public:
    /**
     *@param A: A list of integers
     *@param elem: An integer
     *@return: The new length after remove
     */
    int removeElement(vector<int> &A, int elem) {
        for(auto it = A.begin(); it != A.end();) {
            if(*it == elem) {
                it = A.erase(it);
            } else {
                ++it;
            }
        }
        return A.size();
    }
};
